core/host.go

```diff
@@ -61,7 +61,8 @@ func (host *Host) Directory(ctx context.Context, gw bkgw.Client, dirPath string,
 	pipelineName := fmt.Sprintf("%s %s", pipelineNamePrefix, absPath)
 	ctx, subRecorder := progrock.WithGroup(ctx, pipelineName, progrock.Weak())
 
-	localID := fmt.Sprintf("host:%s", absPath)
+	sessionID := gw.BuildOpts().SessionID
+	localID := fmt.Sprintf("host:%s:%s", sessionID, absPath)
```

Comment:
Snuck this change in along the way, but it's mostly a guess. Seems like we would want the concurrent deduping to factor the session ID in, too.